### ISSUE TITLE
Bug: orchestration client fails to parse stack details, creation_time format is wrong 

### ISSUE THREAD

**Issue by taimir** [gophercloud/gophercloud#185]
**Environment:** DevStack, `stable/newton` branch or PackStack, newton release
**gophercloud version:** HEAD of master, 12f19e5e04d617182cffa5c11f189ef0013b9791
**Observed behavior:**
I am using the gophercloud SDK to build a k8s cluster-autoscaler  based on Heat resource groups. Currently, when I try to *extract* the details of a stack result after a GET request, the function fails, complaining that the creation_time field cannot be parsed correctly. Looking at [this line](https://github.com/gophercloud/gophercloud/blob/master/results.go#L208) the current format is:

```{golang}
const RFC3339NoZ = "2006-01-02T15:04:05"
```

This is in line with what I see on the [official HEAT API page](http://developer.openstack.org/api-ref/orchestration/v1/index.html?expanded=show-stack-details-detail), but still it does not work with the Newton installation. When I check the stack creation time with the python heat client, it shows (e.g.):

```
2016-11-07T08:57:42Z
```

So there is a Z in the end. Changing the gophercloud code to:

```{golang}
const RFC3339 = "2006-01-02T15:04:05Z"
```

fixed the issue for me. Maybe the official doc has not been updated?

**Expected behaviour**:
 The *orchestration* client of gophercloud should not fail to extract the information for a stack result in OpenStack newton.

**Comment by scott-chaney** [comment]
Looks like [This commit](https://github.com/openstack/heat/commit/f655726560bd59cc7d3e18af1d8d53b14c007a7d) added the Zone offset to some stack timestamps.  We are running into the same issue when upgrading from Mitaka.

@taimir did you find another workaround? This issue is from a while ago and I am curious as to why a fix wasn't submitted.  I am happy into looking into a backwards compatible solution but I want to check to see if you ran into anything that prevented you from moving forward with a fix.

**Comment by jtopjian** [comment]
@scott-chaney Nice find - thank you!

This PR (https://github.com/gophercloud/gophercloud/pull/1177) was going to change everything back to the standard RFC 3339 format, but given the link you gave, we will need to support both formats.

**Comment by ricolin** [comment]
@scott-chaney Thanks for clean the history, I will put both support in new patch

**Comment by ricolin** [comment]
@jtopjian So what's the supporting policy for gophercloud? Forever, As long as we can, or something else?

**Comment by jtopjian** [comment]
> So what's the supporting policy for gophercloud? Forever, As long as we can, or something else?

That's a good question that doesn't really have an easy answer.

In my opinion, Heat should have never changed the time format back and forth in the middle of an API version. While that might not have caused breaking changes to a language like Python, it certainly causes problems for a typed language like Go. So Gophercloud must now support both time formats for as long as the v1 API is around.

But that's not realistic because it adds a lot of maintenance overhead to the project. After a few years, OpenStack clouds would have upgraded beyond the problematic releases, so Gophercloud can, at some point, drop the support for both formats.

There's ongoing discussion about adding versioning to Gophercloud. Once that happens, we can more easily add and drop changes like this. For now, though, we need to support both formats.

**Comment by jtopjian** [comment]
Fixed in #1221
